src/Controller/Webhook/WebhookBase.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Boekuwzending\Magento\Controller\Webhook;
+
+use Boekuwzending\Magento\Utils\Constants;
+use JsonException;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\ScopeInterface;
+use Psr\Log\LoggerInterface;
+
+abstract class WebhookBase
+{
+    protected const HMAC_HEADER_NAME = 'Hmac';
```

Comment:
is het niet logischer hier een X-HEADER-HMAC o.i.d. voor te gebruiken?

Reply:
Nou ja, een webhook-body heeft al een meta-element:

```
{
  "meta": {
    "event_name": "webhook:label:created",
    "date_created": "2021-07-30T12:56:19+00:00",
    "hmac": "3156c27055658e5f5f97caa3a302d5733b8544aedfd15dd94115751104adc671"
  },
  "data": {
    "entity_id": "886cc004-f135-11eb-91a8-0242ac140004",
    "shipment_id": "883e46f2-f135-11eb-9826-0242ac140004",
    "external_order_id": "d976f34e-f134-11eb-9d50-0242ac140004",
    "carrier_code": "buzzie",
    "carrier_title": "Buzzie",
    "tracking_number": "1000028028993"
  }
}
```

Dus daar kunnen we het mooi in kwijt. Het data-element wordt gebruikt voor de HMAC. Er is niks tegen op een header, maar we kunnen het net zo goed hierin doen. Wat vind jij?

Reply:
prima voor nu